main/src/ca/uwaterloo/flix/language/phase/unification/AssocTypeSubstitution.scala

```diff
@@ -0,0 +1,175 @@
+/*
```

Comment:
I would like an introduction to these classes at some point.

Reply:
Is this going to replace Substitution?

Reply:
Maybe a more general form could be used, yes.